### DISCUSSION TITLE
xmlsec & Android boringssl

### DISCUSSION THREAD

**Discussion by marcocacre** [lsh123/xmlsec#763] [Q&A] [answered]
Hi all, i tried to integrate and compile xmlsec into the Android AOSP BSP (version 13)  without success. Do you have any plan to add support to boringssl ? or could you give me an advice to found an alternative in Android. 
Thanks a lot for your kindly reply
BR
Marco

**Comment by lsh123** [comment]
Can you share the errors you get? in theory boringssl should be supported but since Google doesn't provide any guarantees about stability of the API, it's hard to say if it actually works or not at any given moment.

**Comment by marcocacre** [comment]
Hi I attached the error log and the Android.bp I'm using. 
I'm not sure if these are all the errors that will occurs since the compilation in Android stop after few errros.
I got these errors after some little fix and silent some warning in compilation that the android compiler consider as error.
thanks!

[error.txt](https://github.com/lsh123/xmlsec/files/14348615/error.txt)
[Android.bp.txt](https://github.com/lsh123/xmlsec/files/14348634/Android.bp.txt)

**Comment by lsh123** [comment, marked as answer]
Thanks, I created issue #764, let's continue discussion there. This specific error is trivial to fix -- it's just a different signature for the function ("const X509_NAME_ENTRY *const *" vs "const struct X509_name_entry_st **"). I just wish that BoringSSL wouldn't make API changes like this all the time.
